projects/libs/core-library-typescript/src/block-storage/block-storage.ts

```diff
@@ -59,56 +65,52 @@ export class BlockStorage {
   public async addBlock(block: types.Block) {
     this.verifyNewBlock(block);
 
-    await this.db.put<number>(BlockStorage.LAST_BLOCK_ID_KEY, block.header.id);
+    await this.db.put<number>(BlockStorage.LAST_BLOCK_HEIGHT_KEY, block.header.height);
     await this.putBlock(block);
 
     this.lastBlock = block;
 
-    logger.info(`Added new block with block ID: ${block.header.id}`);
+    logger.info(`Added new block with block height: ${block.header.height}`);
   }
 
   // Returns an array of blocks, starting from a specific block ID and up to the last block.
-  public async getBlocks(fromLastBlockId: number): Promise<types.Block[]> {
+  public async getBlocks(fromLastBlockHeight: number): Promise<types.Block[]> {
     const blocks: types.Block[] = [];
 
-    if (fromLastBlockId == this.lastBlock.header.id) {
-      return blocks;
-    }
-
-    for (let i = fromLastBlockId; i < this.lastBlock.header.id; ++i) {
+    for (let i = fromLastBlockHeight; i < this.lastBlock.header.height; ++i) {
       blocks.push(await this.getBlock(i + 1));
     }
 
     return blocks;
   }
 
-  public async getLastBlockId(): Promise<number> {
-    return this.lastBlock.header.id;
+  public async getLastBlock(): Promise<types.Block> {
+    return this.lastBlock;
   }
 
-  public async hasNewBlocks(fromLastBlockId: number): Promise<boolean> {
-    return await this.getLastBlockId() > fromLastBlockId;
+  public async hasNewBlocks(fromLastBlockHeight: number): Promise<boolean> {
+    return (await this.getLastBlock()).header.height > fromLastBlockHeight;
   }
 
   private verifyNewBlock(block: types.Block) {
     if (block.header.version !== 0) {
       throw new Error(`Invalid block version: ${block.header.version}!`);
     }
 
-    if (block.header.id !== this.lastBlock.header.id + 1) {
-      throw new Error(`Invalid block ID of block: ${JSON.stringify(block)}!`);
+    if (block.header.height !== this.lastBlock.header.height + 1) {
+      throw new Error(`Invalid block height of block: ${JSON.stringify(block)}!`);
     }
 
-    if (block.header.prevBlockId !== this.lastBlock.header.id) {
-      throw new Error(`Invalid prev block ID of block: ${JSON.stringify(block)}! Should have been ${this.lastBlock.header.id}`);
+    if (!block.header.prevBlockHash.equals(this.lastBlock.hash)) {
+      throw new Error(`Invalid prev block hash of block: ${JSON.stringify(block)}! Should have been ${JSON.stringify(this.lastBlock.hash)}`);
     }
```

Comment:
I don't think we should have a hash in the block at all, but if we will - its verifisation should be here as well.